components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -4469,4 +4477,84 @@ public static String resolveTenantDomain(HttpServletRequest request) {
         }
         return IdentityTenantUtil.getTenantDomainFromContext();
     }
+
+    /**
+     * Get user role list from federated user attributes.
+     * Used in OIDC flow.
+     *
+     * @param userAttributes User attribute
+     * @return user role-list
+     */
+    public static List<String> getRolesFromFederatedUserAttributes(Map<ClaimMapping, String> userAttributes) {
+
+        Optional<ClaimMapping> roleClaimMapping = Optional.ofNullable(userAttributes).get().entrySet().stream()
+                .map(entry -> entry.getKey())
+                .filter(claim -> StringUtils.equals(FEDERATED_ROLE_CLAIM_URI, claim.getRemoteClaim().getClaimUri()))
```

Comment:
check the `local claim uri`

Reply:
there is no mapping for local claim uri here ie we are using the same URI for remote & local claim https://github.com/wso2/carbon-identity-framework/blob/master/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticatedUser.java#L110

Reply:
currently, we are targetting OIDC flow only (yet to expand this to other protocols) & use `EnterpriseLoginAuthenticator` as the recommended IDP which also uses the OIDC protocol. To make it general we need to rely on IDP claim configuration data (not the application data)from the app tenant. This will introduce an additional DB call & we have a limitation in getting the correct IDP name from the authenticated user for this particular scenario

Reply:
@senthalan